### PR TITLE
test: add property-based fuzz tests for content parsing

### PR DESCRIPTION
**Pull request by danielmarv**
# Description
This pull request introduces property-based fuzz testing for the content parsing logic, adds the necessary dependencies to support fuzz testing, and updates the package lockfile to improve platform compatibility for various native dependencies.

This should improve the Scorecard Fuzzing signal because Scorecard recognizes JavaScript/TypeScript property-based fuzzing through fast-check.

**Testing improvements:**

* Added a comprehensive fuzz test suite (`content-parsing.fuzz.test.ts`) using `fast-check` to validate the robustness of content parsing utilities against arbitrary and edge-case inputs, ensuring safer handling of malformed or unexpected frontmatter and markdown.
* Added the `fast-check` library as a development dependency in `package.json` and updated `pnpm-lock.yaml` accordingly. [[1]](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R43) [[2]](diffhunk://#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR83-R85) [[3]](diffhunk://#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR1564-R1567) [[4]](diffhunk://#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR4423-R4426) [[5]](diffhunk://#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR2380-R2382) [[6]](diffhunk://#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR5381-R5382)

**Dependency and platform compatibility updates:**

* Updated multiple native and binary package entries in `pnpm-lock.yaml` to explicitly include the `libc` field, improving compatibility and clarity for different Linux distributions (glibc vs musl) across architectures. [[1]](diffhunk://#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR352-R450) [[2]](diffhunk://#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR524-R545) [[3]](diffhunk://#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR613-R648) [[4]](diffhunk://#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR726-R747) [[5]](diffhunk://#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR964-R1006) [[6]](diffhunk://#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR2026-R2047)

**Configuration cleanup:**

* Removed the `fuzzing` annotation from `.scorecard.yml` as it is not applicable to this documentation-only repository.

## Changes Made
- [ ] Added...
- [ ] Modified...
- [ ] Fixed...

## Checklist
- [ ] Tests added/updated
- [ ] Documentation updated
- [ ] Linting passes
- [ ] Branch up-to-date with main

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive property-based fuzz tests for content parsing to validate metadata handling, output stability, and safe behavior on malformed content.

* **Chores**
  * Added a property-based testing library to devDependencies.
  * Updated repository check configuration to remove the fuzzing check from annotations.

[Review Change Stack](https://app.coderabbit.ai/change-stack/hiero-ledger/hiero-website/pull/464?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: src/lib/__tests__/content-parsing.fuzz.test.ts [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+}
+
+describe("content parsing fuzz tests", () => {
+  beforeEach(() => {
+    resetFsMocks();
+  });
+
+  it("fuzzes arbitrary markdown and YAML frontmatter through simple page parsing", () => {
+    fc.assert(
+      fc.property(fc.string({ maxLength: 1500 }), raw => {
```
Comment (aceppaluni) [inline src/lib/__tests__/content-parsing.fuzz.test.ts]: Should we switch to ```fc.unicodeString()``` to get better parser coverage?

File: src/lib/__tests__/content-parsing.fuzz.test.ts [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+              slug: expect.any(String),
+              title: expect.any(String),
+              date: expect.any(String),
+              featuredImage: expect.any(String),
+              authors: expect.any(Array),
+              categories: expect.any(Array),
+              tags: expect.any(Array),
+            }),
+          );
+          expect(Number.isNaN(Date.parse(post.date))).toBe(false);
```
Comment (aceppaluni) [inline src/lib/__tests__/content-parsing.fuzz.test.ts]: Does this hide timezone bugs?

**Review by aceppaluni — CHANGES_REQUESTED** [review]

File: src/lib/__tests__/content-parsing.fuzz.test.ts (hunk 2 above)
Comment (danielmarv) [reply src/lib/__tests__/content-parsing.fuzz.test.ts]: Good catch. `Date.parse()` may be too weak here since it can hide timezone/normalization issues.

I’ll update the test to assert that the parsed date is valid and remains in canonical UTC ISO format.

File: src/lib/__tests__/content-parsing.fuzz.test.ts (hunk 1 above)
Comment (danielmarv) [reply src/lib/__tests__/content-parsing.fuzz.test.ts]: Since this PR is using `fast-check@4.8.0`, I’ll avoid `fc.unicodeString()` because the unicode string arbitraries were deprecated in 3.x and removed in 4.x. I’ll update this to use the current API instead, most likely `fc.string({ unit: "binary", maxLength: 1500 })`, so the fuzz test covers a wider range of Unicode/code-point inputs while staying compatible with fast-check v4.

File: src/lib/__tests__/content-parsing.fuzz.test.ts [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+}
+
+describe("content parsing fuzz tests", () => {
+  beforeEach(() => {
+    resetFsMocks();
+  });
+
+  it("fuzzes arbitrary markdown and YAML frontmatter through simple page parsing", () => {
+    fc.assert(
+      fc.property(fc.string({ unit: "binary", maxLength: 1500 }), raw => {
```
Comment (aceppaluni) [inline src/lib/__tests__/content-parsing.fuzz.test.ts]: For lines that contain '''fc.string({unit: "binary"})''' would it be good idea to add something like
'''
fc.record({
  frontmatter: ...
  markdownSections: ...
  headings: ...
  codeBlocks: ...
})
'''
To exercise things like parser boundaries, delimiter confusion, metadata extraction logic and, markdown edge cases?

File: src/lib/__tests__/content-parsing.fuzz.test.ts [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+        numRuns: FUZZ_RUNS,
+        seed: 20260527,
+      },
+    );
+  });
+
+  it("reports arbitrary TOML parser failures as Error instances", () => {
+    fc.assert(
+      fc.property(fc.string({ unit: "binary", maxLength: 500 }), input => {
+        try {
```
Comment (aceppaluni) [inline src/lib/__tests__/content-parsing.fuzz.test.ts]: This seems like it is testing the TOML library doesn’t throw odd non-Error values. 

Should this instead assert wrapper behavior in your the layer?

**Review by aceppaluni — CHANGES_REQUESTED** [review]

**Review by aceppaluni — APPROVED** [review]
Great work!
